### PR TITLE
chore(replays): Move delete replays script into the sentry repo

### PR DESCRIPTION
**Pull request by cmanallen**
The deletion **_script_** currently exists in our private repo.  Changing the delete **_task_** becomes tricky because the public and private repos depend on one another.  To fix this, I'm moving the delete **_script_** into sentry and will be calling it in the private repo.  This means we can modify the deletion script and the deletion task in the same PR.

### CODE REVIEW

File: tests/sentry/replays/scripts/test_delete_replays.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+)
+from sentry.testutils.cases import ReplaysSnubaTestCase
+from sentry.utils.json import dumps_htmlsafe
+
+
+class TestDeleteReplays(ReplaysSnubaTestCase):
+    def store_replay_segments(
+        self,
+        replay_id: str,
+        project_id: str,
```
Comment (aliu3ntry) [inline tests/sentry/replays/scripts/test_delete_replays.py:27]: I merged a small PR in getsentry yesterday fixing the type of this parameter + line 36. Could you include these changes? https://github.com/getsentry/getsentry/pull/13327

Comment (cmanallen) [reply tests/sentry/replays/scripts/test_delete_replays.py:27]: Oof.  Forgot to pull before I copied it over.  Thanks I've updated it :D

Comment (aliu3ntry) [reply tests/sentry/replays/scripts/test_delete_replays.py]:
```suggestion
        project_id: int,
```

**Review by bruno-garcia — APPROVED** [review]

File: tests/sentry/replays/scripts/test_delete_replays.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    mock_segment_console,
+    mock_segment_fullsnapshot,
+    mock_segment_init,
+    mock_segment_nagivation,
+)
+from sentry.testutils.cases import ReplaysSnubaTestCase
+from sentry.utils.json import dumps_htmlsafe
+
+
+class TestDeleteReplays(ReplaysSnubaTestCase):
```
Comment (aliu3ntry) [inline tests/sentry/replays/scripts/test_delete_replays.py:23]: this is for a separate PR, but I know from the [threadpool PR](https://github.com/getsentry/sentry/pull/66881) we'll eventually need `TransactionTestCase` here

**Review by aliu39 — APPROVED** [review]

**Review by ryan953 — APPROVED** [review]

**Pull request merged**